tests/loops/epoch/test_training_epoch_loop.py

```diff
@@ -141,3 +145,29 @@ def test_prepare_outputs_training_batch_end_manual(batch_end_outputs, expected):
         num_optimizers=-1,  # does not matter for manual optimization
     )
     assert prepared == expected
+
+
+def test_no_val_on_train_epoch_loop_restart(tmpdir):
+    """Test that training validation loop doesn't get triggered at the beginning of a restart."""
+    trainer_kwargs = {
+        "max_epochs": 1,
+        "limit_train_batches": 1,
+        "limit_val_batches": 1,
+        "num_sanity_val_steps": 0,
+        "enable_checkpointing": False,
+    }
+    trainer = Trainer(**trainer_kwargs)
+    model = BoringModel()
+    trainer.fit(model)
+    ckpt_path = str(tmpdir / "last.ckpt")
+    trainer.save_checkpoint(ckpt_path)
+
+    trainer_kwargs["max_epochs"] = 2
+    trainer = Trainer(**trainer_kwargs)
+
+    with patch.object(
```

Comment:
Hi, @rohitgr7, could we also check if `val_loop.advance()` is called as well? One of our internal users is trying to patch your fix to their codebase, but encounter `val_loop` runs with  zero data reading. 
Because `val_loop.run()` does not mean we indeed have evaluation, but we may just skip the loop within `val_loop` due to `self.done=True`. Thanks a lot.

Reply:
sure!

Reply:
sure!

Reply:
Having added the `advance` mock, the `run` mock becomes redundant

Reply:
we can't patch 2 methods together?

Reply:
should I remove it?

Reply:
We can remove `run` mock. If we mock `advance`, then there's no need to check `run`. because if `advance` is called, so is `run`.